spec/src/main/asciidoc/core/invokers.asciidoc

```diff
@@ -0,0 +1,181 @@
+[[method_invokers]]
+== Method invokers
+
+CDI-based frameworks often need to invoke application methods declared on managed beans.
+Frameworks cannot invoke application methods directly, because they are not compiled against the application code.
+However, during application deployment, frameworks may observe application methods through CDI extensions and build an `Invoker` for each relevant method.
+The invokers can then be used at application runtime to invoke the methods indirectly.
+
+Method invokers are not supposed to be used by application code, as applications may invoke their own methods directly.
+
+[[building_invoker]]
+=== Building an `Invoker`
+
+The CDI container allows building an `Invoker` for a method of an <<enablement,enabled>> <<managed_beans,managed bean>>.
+The method for which the invoker is built is called the _target method_ of the invoker, and the managed bean is called the _target bean_ of the invoker.
+
+Invalid target methods are:
+
+* private methods,
+* constructors,
+* methods declared on the `java.lang.Object` class, except of the `toString()` method,
+* methods that are not declared on the bean class of the target bean or inherited from its supertypes.
+
+Attempting to build an invoker for an invalid target method leads to a deployment problem.
+
+When the target bean is not a managed bean, attempting to build an invoker leads to a deployment problem.
+When the target bean is an interceptor, attempting to build an invoker leads to a deployment problem.
+
+Multiple managed beans may inherit a method from a common supertype.
+In that case, an invoker must be built for each target bean individually.
+An invoker built for one target bean may not be used to invoke the target method on an instance of another target bean.
+
+The only way to build an invoker is using the `InvokerBuilder`.
+An `InvokerBuilder` can only be obtained in CDI portable extensions and build compatible extensions.
+See <<invoker_builder>> for more information.
+
+[[using_invoker]]
+=== Using an `Invoker`
+
+The `Invoker` interface contains a single method:
+
+[source,java]
+----
+public interface Invoker<T, R> {
+    R invoke(T instance, Object[] arguments);
+}
+----
+
+This section describes the behavior of the `invoke()` method.
```

Comment:
This sentence seems superfluous as the intention of the text below is rather obvious.

Reply:
Fair enough. At some point, it seemed necessary to me, but I'm probably overly cautious here.

Reply:
I made this a heading of a subsection that specifies all the boring but necessary details.